content/src/service/synchronization/SynchronizationManager.ts

```diff
@@ -93,6 +119,8 @@ export class ClusterSynchronizationManager implements SynchronizationManager {
         return deploymentStream.pipe(sourceData)
       })
 
+      this.timeoutStreams(streams)
+
       // Process them together
       await this.deployer.processAllDeployments(streams)
```

Comment:
I'd move the timeout logic inside `this.deployer.processAllDeployments`. That should handle the timeout of the received streams. This file may be too high level